src/librustc/metadata/decoder.rs

```diff
@@ -233,10 +220,16 @@ fn get_provided_source(d: rbml::Doc, cdata: Cmd) -> Option<ast::DefId> {
     })
 }
 
-fn each_reexport<F>(d: rbml::Doc, f: F) -> bool where
+fn each_reexport<F>(d: rbml::Doc, mut f: F) -> bool where
```

Comment:
Perhaps this could return a `TaggedDocsIterator` instead of continuing to use a closure?